### PR TITLE
Feature/db connector

### PR DESCRIPTION
**Pull request by VasylKosiv**

### CODE REVIEW

File: storage/postgres.go [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+	"database/sql"
+	"fmt"
+	"log"
+
+	// _ mean that we can use all function from this package
+	_ "github.com/lib/pq"
+)
+
+// Config of Postgres DB
+type PgClient struct {
```
Comment (crnopster) [inline storage/postgres.go]: WDYT about renaming PgClient to DBConfig or Config

Comment (VasylKosiv) [reply storage/postgres.go]: Agree, that's better name. Changed it.

File: storage/postgres.go [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
+// Package storage has to provide an interface to create a new postgres client
+// with ability to reconnect after connection failure.
 package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	// _ mean that we can use all function from this package
```
Comment (alpaka-alpachino) [inline storage/postgres.go]: Do you need it here?

Comment (VasylKosiv) [reply storage/postgres.go]: Yes, I think so. This is massage from linter, without this comment: "A blank import should be only in a main or test package, or have a comment justifying it."

Comment (crnopster) [reply storage/postgres.go]: for this case you can use another type of comments
//nolint:{linterName}

Comment (VasylKosiv) [reply storage/postgres.go]: I think that my comment is more informative.

File: cmd/umserver/main.go [hunk 3]
```diff
@@ -43,8 +45,21 @@ func main() {
...
+		User:     "postgres",
+		Password: "postgres",
+		DBName:   "um_db",
+	}
+
+	db, err := storage.ConnectToDB(&pgConfig)
+	if err != nil {
+		log.Print(err)
+	}
+	defer db.Close()
```
Comment (dmitriyminer) [inline cmd/umserver/main.go]: `db` can be `nil`

File: storage/postgres.go (hunk 2 above)
Comment (radikh) [reply storage/postgres.go]: This comment says absolutely opposite to real behavior.

Comment (VasylKosiv) [reply storage/postgres.go]: My falt. Canged it.

File: cmd/umserver/main.go (hunk 3 above)
Comment (VasylKosiv) [reply cmd/umserver/main.go]: Unfortunately I can't understand what your comment means. Couldn't describe in more detail. Thank you

[the remark this answers is not in the document]
Comment (VasylKosiv) [reply cmd/umserver/main.go]: Done

File: storage/postgres.go [hunk 4]
```diff
@@ -1,7 +1,45 @@
...
+
+const pgStr = "host=%v port=%v user=%v password=%v dbname=%v sslmode=disable"
+
+//DB connector
+func ConnectToDB(pg *DBConfig) (*sql.DB, error) {
+	pgConfig := fmt.Sprintf(pgStr, pg.Host, pg.Port, pg.User, pg.Password, pg.DBName)
+
+	database, err := sql.Open("postgres", pgConfig)
+	if err != nil {
+		log.Print("Could not connect to ", pg.DBName)
```
[the remark this answers is not in the document]
Comment (VasylKosiv) [reply storage/postgres.go]: Done

File: storage/postgres_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		},
+	}
+	for _, test := range tests {
+		_, get := ConnectToDB(test.config)
+
+		if (get != nil) == test.expect {
+			t.Errorf("ConnectToDB(%v) expect %v", test.name, test.expect)
+			return
+		}
+	}
```
Comment (radikh) [inline storage/postgres_test.go]: Please use assertion library for test checks. github.com/stretchr/testify will be good enough.

Comment (VasylKosiv) [reply storage/postgres_test.go]: Done

File: storage/postgres.go [hunk 6]
```diff
@@ -16,3 +19,23 @@ type DBConfig struct {
...
 	DBName   string
 }
+
+const pgStr = "host=%v port=%v user=%v password=%v dbname=%v sslmode=disable"
+
+//DB connector
+func ConnectToDB(pg *DBConfig) (*sql.DB, error) {
+	pgConfig := fmt.Sprintf(pgStr, pg.Host, pg.Port, pg.User, pg.Password, pg.DBName)
+
+	database, err := sql.Open("postgres", pgConfig)
```
Comment (radikh) [inline storage/postgres.go:29]: Move "postgres" to contants.

File: storage/postgres.go [hunk 7]
```diff
@@ -16,3 +19,23 @@ type DBConfig struct {
 	Password string
 	DBName   string
 }
+
+const pgStr = "host=%v port=%v user=%v password=%v dbname=%v sslmode=disable"
+
+//DB connector
+func ConnectToDB(pg *DBConfig) (*sql.DB, error) {
+	pgConfig := fmt.Sprintf(pgStr, pg.Host, pg.Port, pg.User, pg.Password, pg.DBName)
```
Comment (radikh) [inline storage/postgres.go:27]: Move sprintf to a separate function.

File: storage/postgres.go (hunk 6 above)
Comment (VasylKosiv) [reply storage/postgres.go]: Done

File: storage/postgres.go (hunk 7 above)
Comment (VasylKosiv) [reply storage/postgres.go]: Done

File: storage/postgres.go [hunk 8]
```diff
@@ -16,3 +19,28 @@ type DBConfig struct {
 	Password string
 	DBName   string
 }
+
+const pgStr = "host=%v port=%v user=%v password=%v dbname=%v sslmode=disable"
+const DBDriverName = "postgres"
+
+//DB connector
+func ConnectToDB(pg *DBConfig) (*sql.DB, error) {
```
Comment (dmitriyminer) [inline storage/postgres.go:27]: Comment is a complete sentence that begins with the name of the element it describes 
https://blog.golang.org/godoc

File: storage/postgres.go [hunk 9]
```diff
@@ -16,3 +19,28 @@ type DBConfig struct {
 	Password string
 	DBName   string
 }
+
+const pgStr = "host=%v port=%v user=%v password=%v dbname=%v sslmode=disable"
+const DBDriverName = "postgres"
```
Comment (dmitriyminer) [inline storage/postgres.go:24]: Do you really need this constant as expected?

File: storage/postgres.go [hunk 10]
```diff
@@ -16,3 +19,28 @@ type DBConfig struct {
...
+	err = database.Ping()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return database, nil
+}
+
+func GetDBConfigString(pg *DBConfig) string {
```
Comment (dmitriyminer) [inline storage/postgres.go:44]: Helper function should be unexported

File: storage/postgres.go [hunk 11]
```diff
@@ -16,3 +19,28 @@ type DBConfig struct {
...
+//DB connector
+func ConnectToDB(pg *DBConfig) (*sql.DB, error) {
+	pgConfig := GetDBConfigString(pg)
+	database, err := sql.Open(DBDriverName, pgConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	err = database.Ping()
```
Comment (dmitriyminer) [inline storage/postgres.go:35]: Do not separate error handling block
```
	err = database.Ping()
	if err != nil {
		return nil, err
	}
```

File: migrator/init/user_for_api.sh [hunk 12]
```diff
@@ -1,2 +1,2 @@
 #!/bin/bash
```
Comment (dmitriyminer) [inline migrator/init/user_for_api.sh:1]: Changed permissions 100644 → 100755
Why?

File: cmd/umserver/main.go [hunk 13]
```diff
@@ -70,8 +72,22 @@ func main() {
 	}()
 	logger.LogUM.Info("Server Listening at %s...", srv.Addr)
 
-	// TODO: There will be actual information about PostgreSQL connection in future
-	// ...
+	dbConfig, err := cfg.DBConfig(ctx)
+	if err != nil {
+		log.Fatal(err)
```
Comment (dmitriyminer) [inline cmd/umserver/main.go:77]: It would be great to add context about error

File: cmd/umserver/main.go [hunk 14]
```diff
@@ -70,8 +72,22 @@ func main() {
...
+	}
+
+	db, err := storage.ConnectToDB(dbConfig)
+
+	if err != nil {
+		logger.LogUM.Error("%v\n", err)
```
Comment (dmitriyminer) [inline cmd/umserver/main.go:83]: Please add error message

File: storage/postgres.go (hunk 8 above)
Comment (VasylKosiv) [reply storage/postgres.go:27]: Fixed it

File: storage/postgres.go (hunk 9 above)
Comment (VasylKosiv) [reply storage/postgres.go]: Actually no, fixed it

File: storage/postgres.go (hunk 10 above)
Comment (VasylKosiv) [reply storage/postgres.go]: Fixed it

File: storage/postgres.go (hunk 11 above)
Comment (VasylKosiv) [reply storage/postgres.go:36]: My linter asked me for that. Fixed it.

File: migrator/init/user_for_api.sh (hunk 12 above)
Comment (VasylKosiv) [reply migrator/init/user_for_api.sh:1]: Didn't do that. Can't imagine why it happened

File: cmd/umserver/main.go (hunk 13 above)
Comment (VasylKosiv) [reply cmd/umserver/main.go]: Done

File: cmd/umserver/main.go (hunk 14 above)
Comment (VasylKosiv) [reply cmd/umserver/main.go]: Done

File: migrator/init/user_for_api.sh (hunk 12 above)
Comment (VasylKosiv) [reply migrator/init/user_for_api.sh:1]: Changed premission to 644

**Pull request merged**